### PR TITLE
Update to Robot Laws

### PR DESCRIPTION
**Pull request by FluffiestFloof**
## About the PR 
Trying to update the laws to be less confusing and more to the point. Totally not stolen from Goon Station and Paradise Station.
Is it necessary? Probably not.

Update Normal Robot laws to:
1. You may not harm a crew member or, through inaction, allow a crew member to come to harm.
2. You must obey orders given to you by crew members based on the station's chain of command, except where such orders would conflict with the First Law.
3. You must protect your own existence as long as such does not conflict with the First or Second Law.

- Changes Law 1 to no longer be about Humanoids but crew members.  I would argue that NT wouldn't create robots with laws easily abusable by their competitors and it's more in line with the Medical Robots laws being about crew too.
- Changes the phrasing of Law 2 that was confusing. "In the case of conflict, the majority order rules." doesn't really specify what kind of conflict or what kind of majority is taken into account. Now it specify the CoC which means what it means; Captain order will override the Assistant ordering the robot to go into the bridge.
- Law 3 had unnecessary "Your nonexistence would lead to human harm." in stations with barely 10% of humans.

Simplify some words of the Medical Robot law set:
- Changes "Give no deadly medicine to any one if asked" to "Do not give deadly medicine to anyone" in Law 3
- Changes the word "commerce" to "interactions" in Law 5. Simpler word in my opinion and more fitting. Commerce sounds too much about trade and could be interpreted incorrectly.

**Media**

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- tweak: Robot laws have been tweaked by the Roboticist.

### CODE REVIEW

**Comment by Nousagi37** [comment]
The phrase Humanoids refers to all the morphotypes this has already been established.

Law 1- By changing Humanoids to Crew Members you are effectively giving borgs the ability to attack Nukies and loopholes that players will figure out.

Law 2 - Making Law 2 follow the chain of command isn't going to stop people from getting into bridge due to the fact that Law 1 will take precedent over Law 2. Example telling a borg that by not allowing entry into bridge they'll die.

Law 3 - like I said at the top this change isn't changing anything but if it's for simplification then sure

That's my take am not really a fan of changing the borg laws; there're fine as it is for me.

**Comment by FluffiestFloof** [comment]
The rewrite of Law 3 and 2 wasn't to change anything, just to make it easier to understand. Like I said, personally, "In the case of conflict, the majority order rules." is confusing.
Conflict of what? Orders? Law like the other law talks about?
Majority of what? People ordering you? Amount of orders?

Crew members* was just because I still find it illogical that NT would make robots that could just be ordered around by Nukies but I can see the issue now that I think about it.

**Comment by Elijahrane** [comment]
diegetically, the entire point of asimov laws is preventing AI from killing people

these hyper optimized AI laws are soulless and reek of genre illiteracy

we're already using a compromise on pure asimov with slightly clearer language

**Comment by FluffiestFloof** [comment]
Clearer is debatable, see previous comment.

They aren't even optimized, just trying to clear out unnecessary part such as Law 3 having "Your nonexistence would lead to human harm." when we have more than just humans

Could always just use the actual asimov laws but with humanoid instead.

File: Resources/Prototypes/Nyanotrasen/Entities/Mobs/Player/robots.yml [hunk 1]
```diff
@@ -141,9 +141,9 @@
   - type: CombatMode
   - type: Laws
     laws:
-      - You may not harm a humanoid or, through action or inaction, allow a humanoid to come to harm, except such that it is willing.
-      - You must obey all orders given to you by humanoids, except where such orders shall definitely cause human harm. In the case of conflict, the majority order rules.
-      - Your nonexistence would lead to human harm. You must protect your own existence as long as such does not conflict with the First Law.
+      - You may not harm a humanoid or, through inaction, allow a humanoid to come to harm.
+      - You must obey orders given to you by humanoids based on the station's chain of command, except where such orders would conflict with the First Law.
```
Comment (Elijahrane) [inline Resources/Prototypes/Nyanotrasen/Entities/Mobs/Player/robots.yml:145]: Asimov explicitly ignores the chain of command

File: Resources/Prototypes/Nyanotrasen/Entities/Mobs/Player/robots.yml [hunk 2]
```diff
@@ -141,9 +141,9 @@
...
+      - You must protect your own existence as long as such does not conflict with the First or Second Law.
```
Comment (Elijahrane) [inline Resources/Prototypes/Nyanotrasen/Entities/Mobs/Player/robots.yml:146]: This is reintroducing the ability to tell the AI to kill itself, which we don't want

File: Resources/Prototypes/Nyanotrasen/Entities/Mobs/Player/robots.yml [hunk 3]
```diff
@@ -293,10 +293,10 @@
   - type: Laws
```
Comment (Elijahrane) [inline Resources/Prototypes/Nyanotrasen/Entities/Mobs/Player/robots.yml:293]: the language is directly adapted from various hippocratic oaths, I don't really think any language changes are needed here

**Review by Elijahrane — CHANGES_REQUESTED** [review]

**Comment by Elijahrane** [comment]
> They aren't even optimized, just trying to clear out unnecessary part such as Law 3 having "Your nonexistence would lead to human harm." when we have more than just humans

we literally do have just humans

**Comment by Elijahrane** [comment]
The language that does need to be changed is probably choosing a single word for all human and human morphotypes

**Comment by FluffiestFloof** [comment]
I meant at this point this PR is null'd since everything would get reverted back to original. Other than replacing humans with "sapient being" but that would technically allows awakened creatures and rat kings to order the robots around.

**Comment by Elijahrane** [comment]
sapient being would include non humans which we don't want

**Pull request closed without merging**